docker/rucio_client/Dockerfile.loadtest

```diff
@@ -2,6 +2,3 @@ FROM registry.cern.ch/cmsrucio/rucio_client:latest
 
 COPY loadtest /loadtest
 
-ENTRYPOINT ["/loadtest/loadtest.py"]
```

Comment:
Okay, but this ENTRYPOINT won't work as two arguments have no default values in the source code:
```
[root@900dd3eedd39 ~]# /loadtest/loadtest.py
usage: loadtest.py [-h] --source_rse_expression SOURCE_RSE_EXPRESSION --dest_rse_expression DEST_RSE_EXPRESSION [--account ACCOUNT] [--activity ACTIVITY] [--filesize FILESIZE] [-v]
loadtest.py: error: the following arguments are required: --source_rse_expression, --dest_rse_expression
```
Do you want to do the entrypoint as follows?
```
/loadtest/loadtest.py -v   --source_rse_expression ${SOURCE_EXPRESSION} --dest_rse_expression ${DEST_EXPRESSION}
```

Reply:
You can do either way. 
I would prefer to leave that out because that would require specifying env vars like this
`docker run ...... -e SOURCE_EXPRESSION=<src_exp> -e DEST_EXPRESSION=<dest_exp>`

as opposed to 
`docker run loadtest --source_rse_expression <exp> --dest_rse_expression <exp>`

Btw, can you give the bottom one a try with the existing image?

Reply:
The latter one doesn't work. I added an entrypoint and it works

```
$ podman run -d --name loadtest -v "/cvmfs/grid.cern.ch/etc/grid-security:/etc/grid-security/" -v "x509up:/tmp/"  -e RUCIO_HOME="/opt/rucio-prod/" -e SOURCE_EXPRESSION="loadtest=True" -e DEST_EXPRESSION="loadtest=True" -e ACTIVITY="Debug" registry.cern.ch/haozturk/cmsloadtest_hasan:v5
09faece69c1a8219ffde23bd0702a8a1636530e4e385489f50ea191f63dca1a7
$
$ podman logs loadtest
2024-04-24 16:56:02,215 __main__:INFO:No link between T1_IT_CNAF_Tape and T0_CH_CERN_Tape, skipping load test
2024-04-24 16:56:02,539 __main__:INFO:Link between T1_IT_CNAF_Tape and T2_PL_Cyfronet with load test rule 035a4f3664d24be5bb5fa18486b392c7 last updated 53898.539291s ago (target=21600.0), marking destination replicas unavailable
...
```
How does it look now?

Reply:
You shouldn't need this `"/bin/bash", "-c",` , this will open a shell inside a shell

Reply:
This is the "shell form" and as I understand it's the only way to do variable substitution for the arguments. The "exec form" [1] doesn't work because of this. See [2]

[1]
```
["/loadtest/loadtest.py", "-v", "--source_rse_expression", "$SOURCE_EXPRESSION", "--dest_rse_expression", "$DEST_EXPRESSION", "--activity", "$ACTIVITY"]
```
[2] https://stackoverflow.com/questions/37904682/how-do-i-use-docker-environment-variable-in-entrypoint-array

Reply:
Not exactly, what you are doing it is starting a shell in exec form. 
The shell form requires you to not specify a list (format) of commands but a single string. 

Anyways as eric suggested, we can also remove the ENTRYPOINT completely.